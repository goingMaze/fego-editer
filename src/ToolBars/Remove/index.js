import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Modifier, EditorState } from 'draft-js'
import { ToolBarBtn } from '../../utils'
import Iconfont from '../../iconfont'

export default class TextAlign extends Component {
	static propTypes = {
		editorState: PropTypes.object.isRequired,
		onChange: PropTypes.func.isRequired,
	}
	onToggle = () => {
		let { editorState, onChange } = this.props;
		const inlineStyleMap = ['BOLD', 'ITALIC', 'UNDERLINE'];
		let contentState = editorState.getCurrentContent(),
			selection = editorState.getSelection();
		inlineStyleMap.forEach(style => {
			contentState = Modifier.removeInlineStyle(contentState, selection, style)
		})
		onChange(EditorState.push(editorState, contentState, 'change-inline-style'))
	}
	render() {
		return (
			<div className="FegoEditor-toolbars">
				<ToolBarBtn
					onToggle={this.onToggle}
					style='remove'
				>
					{Iconfont['iconRemove']}
				</ToolBarBtn>
			</div>
		)
	}
}