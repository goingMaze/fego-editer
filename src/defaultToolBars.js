const defaultToolBars = {
	options: ['BlockType', 'Color', 'FontFamily', 'FontSize', 'Image', 'Inline', 'Link', 'List', 'TextAlign', 'Video', 'History'],
	BgColor: [],
	BlockType: ['正文', 'H1', 'H2', 'H3', 'H4', 'H5', 'H6'],
	Color: ['rgb(97,189,109)', 'rgb(26,188,156)', 'rgb(84,172,210)', 'rgb(44,130,201)',
		'rgb(147,101,184)', 'rgb(71,85,119)', 'rgb(204,204,204)', 'rgb(65,168,95)', 'rgb(0,168,133)',
		'rgb(61,142,185)', 'rgb(41,105,176)', 'rgb(85,57,130)', 'rgb(40,50,78)', 'rgb(0,0,0)',
		'rgb(247,218,100)', 'rgb(251,160,38)', 'rgb(235,107,86)', 'rgb(226,80,65)', 'rgb(163,143,132)',
		'rgb(239,239,239)', 'rgb(255,255,255)', 'rgb(250,197,28)', 'rgb(243,121,52)', 'rgb(184,49,47)',
		'rgb(209,72,65)', 'rgb(124,112,107)', 'rgb(209,213,216)'],
	FontFamily: [{
		name: 'SimHei', family: '"SimHei", "黑体"'
	}, {
		name: 'Yahei', family: '"Microsoft Yahei", "微软雅黑"'
	}, {
		name: 'KaiTi', family: '"KaiTi", "楷体"'
	}, {
		name: 'FangSong', family: '"FangSong", "仿宋"'
	}],
	FontSize: [16, 20, 30, 40, 50, 70, 100, 200],
	History: ['undo', 'redo'],
	Image: {
		uploadUrl: '',
		uploadCallback: null
	},
	Inline: ['bold', 'italic', 'underline'],
	Link: {},
	List: ['ul', 'ol', 'indent', 'outdent'],
	TextAlign: ['left', 'center', 'right', 'justify'],
	Video: {},
	Remove: {}
};

export default defaultToolBars
