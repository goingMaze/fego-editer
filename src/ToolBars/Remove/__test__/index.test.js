import React from 'react'
import { mount, render } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { EditorState, ContentState, convertFromHTML } from 'draft-js'
import Remove from '..'
import defaultToolBars from '../../../defaultToolBars'

describe('Remove', () => {
	const contentState = ContentState.createFromBlockArray(convertFromHTML('<div>test</div>'))
	const editorState = EditorState.createWithContent(contentState)
	const wrapper = mount(
		<Remove editorState={editorState} onChange={() => { }} />
	);
	it('渲染时开头应该有一个div', () => {
		expect(wrapper.html().startsWith('<div')).toEqual(true)
	})
	it('should have a toolBarBtn child component', () => {
		expect(wrapper.find('ToolBarBtn').length).toBe(1)
	})
	it('快照', () => {
		const rendered = render(
			<Remove editorState={editorState} onChange={() => { }} config={defaultToolBars.Remove} />
		)
		expect(toJSON(rendered)).toMatchSnapshot()
	})
})