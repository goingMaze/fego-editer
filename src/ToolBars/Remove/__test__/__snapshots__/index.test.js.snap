// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remove 快照 1`] = `
<div
  class="FegoEditor-toolbars"
>
  <div
    class="FegoEditor-toolBarBtn "
  >
    <svg
      aria-hidden="true"
      class="icon"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M863.119893 182.650947l-470.546889 0c-52.05661 0-109.093651 42.177612-127.393435 94.205569L99.758028 747.142461c-18.299785 52.02898 9.06547 94.205569 61.122079 94.205569l470.546889 0c52.057633 0 109.093651-42.177612 127.393435-94.205569l165.42154-470.285946C942.541757 224.828558 915.177526 182.650947 863.119893 182.650947zM663.169955 751.590773l-481.762325 0 84.275405-239.590773 481.762325 0L663.169955 751.590773z"
      />
    </svg>
  </div>
</div>
`;
